stl/inc/ranges

```diff
@@ -3340,7 +3343,12 @@ namespace ranges {
 
         // clang-format off
         _NODISCARD constexpr _Iterator<true> begin() const
-            requires input_range<const _Vw> && is_reference_v<_InnerRng<true>> {
+        #ifdef __clang__ // TRANSITION, LLVM-47414

```

Comment:
I think it would be simpler to use the new concept unconditionally.

We have the bug citation with the concept and I does improve readability

Reply:
We generally prefer to keep both the original code and the workaround active when possible to ensure that neither regresses.

Reply:
As CaseyCarter alluded to, this same workaround on [transform_view](https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/ranges#L2236) still has the the compound requirement and the requirement hoisted into it's own concept. 

I decided to match what I assumed was the appropriate convention; should I go ahead and make this change, or leave it matching the transform_view implementation?